File: jmolecules-axon/02-axonstyle-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/domain/api/command/transfer/ReceiveMoneyTransferCommand.java
package org.jmolecules.example.axonframework.domain.api.command.transfer;

import org.jmolecules.architecture.cqrs.annotation.Command;
import org.jmolecules.ddd.annotation.Association;

/**
 * Invoked by the saga, recipient is target account.
 */
@Command(namespace = "axon.bank", name = "ReceiveMoneyTransferCommand")
public record ReceiveMoneyTransferCommand(
  @Association
  String targetAccountId,

  String moneyTransferId,

  int amount
) {

}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/domain/api/command/transfer/ReceiveMoneyTransferCommand.kt
package org.jmolecules.example.axonframework.domain.api.command.transfer

import org.jmolecules.architecture.cqrs.annotation.Command
import org.jmolecules.ddd.annotation.Association
import org.jmolecules.example.axonframework.domain.api.type.AccountId
import org.jmolecules.example.axonframework.domain.api.type.Amount
import org.jmolecules.example.axonframework.domain.api.type.MoneyTransferId

/**
 * Invoked by the saga, recipient is target account.
 */
@Command(namespace = "axon.bank", name = "ReceiveMoneyTransferCommand")
data class ReceiveMoneyTransferCommand(
    @Association
    val targetAccountId: AccountId,
    val moneyTransferId: MoneyTransferId,
    val amount: Amount
)
